app/forms/floating-ip-create.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useMemo } from 'react'
+import { useWatch } from 'react-hook-form'
+import {
+  useNavigate,
+  type LoaderFunctionArgs,
+  type NavigateFunction,
+} from 'react-router-dom'
+
+import {
+  apiQueryClient,
+  useApiMutation,
+  useApiQuery,
+  useApiQueryClient,
+  type FloatingIp,
+  type FloatingIpCreate,
+  type SiloIpPool,
+} from '@oxide/api'
+import { validateIp } from '@oxide/util'
+
+import {
+  DescriptionField,
+  ListboxField,
+  NameField,
+  SideModalForm,
+  TextField,
+} from 'app/components/form'
+import { getProjectSelector, useForm, useProjectSelector, useToast } from 'app/hooks'
+import { pb } from 'app/util/path-builder'
+
+CreateFloatingIpSideModalForm.loader = async ({ params }: LoaderFunctionArgs) => {
+  await apiQueryClient.prefetchQuery('projectIpPoolList', {
+    query: { ...getProjectSelector(params), limit: 1000 },
+  })
+  return null
+}
+
+type CreateSideModalFormProps = {
+  /**
+   * If defined, this overrides the usual mutation. Caller is responsible for
+   * doing a dismiss behavior in onSubmit as well, because we are not calling
+   * the RQ `onSuccess` defined for the mutation.
+   */
+  onSubmit?: (floatingIpCreate: FloatingIpCreate) => void
+  /**
+   * Passing navigate is a bit of a hack to be able to do a nav from the routes
+   * file. The callers that don't need the arg can ignore it.
+   */
+  onDismiss?: (navigate: NavigateFunction) => void
+  onSuccess?: (floatingIp: FloatingIp) => void
+}
+
+export function CreateFloatingIpSideModalForm({
+  onSubmit,
+  onSuccess,
+}: CreateSideModalFormProps) {
+  // Fetch 1000 to we can be sure to get them all. There should only be a few
+  // anyway. Not prefetched because the prefetched one only gets 25 to match the
+  // query table. This req is better to do async because they can't click make
+  // default that fast anyway.
+  const { data: allPools } = useApiQuery('projectIpPoolList', {
+    query: { limit: 1000 },
+  })
```

Comment:
Use `usePrefetchedApiQuery` here and ditch the comment except the first sentence.